refinery/units/formats/macho/machometa.py

```diff
@@ -0,0 +1,350 @@
+from io import BytesIO
+from typing import Dict, List
+
+from ktool import load_image, load_macho_file, Image, MachOFileType
+from ktool.macho import build_version_command, LOAD_COMMAND,  source_version_command, Struct, uint8_t, uint32_t
+from ktool.codesign import Blob, BlobIndex, CSSLOT_CODEDIRECTORY, SuperBlob, swap_32
+
+from refinery.units import Arg, Unit
+from refinery.units.sinks.ppjson import ppjson
+
+
+class CodeDirectoryBlob(Struct):
+    _FIELDNAMES = [
+        "magic",
+        "length",
+        "version",
+        "flags",
+        "hashOffset",
+        "identOffset",
+        "nSpecialSlots",
+        "nCodeSlots",
+        "codeLimit",
+        "hashSize",
+        "hashType",
+        "platform",
+        "pageSize",
+        "spare2"
+    ]
+    _SIZES = [
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint32_t,
+        uint8_t,
+        uint8_t,
+        uint8_t,
+        uint8_t,
+        uint32_t
+    ]
+    SIZE = 44
+
+    def __init__(self, byte_order="little"):
+        super().__init__(fields=self._FIELDNAMES, sizes=self._SIZES, byte_order=byte_order)
+        self.magic = 0
+        self.length = 0
+        self.version = 0
+        self.flags = 0
+        self.hashOffset = 0
+        self.identOffset = 0
+        self.nSpecialSlots = 0
+        self.nCodeSlots = 0
+        self.codeLimit = 0
+        self.hashSize = 0
+        self.hashType = 0
+        self.platform = 0
+        self.pageSize = 0
+        self.spare2 = 0
+
+CS_ADHOC = 0x0000_0002
+
+class machometa(Unit):
+    """
+    Extract metadata from Mach-O files.
+    """
+    def __init__(
+        self, all: Arg('-c', '--custom',
+            help='Unless enabled, all default categories will be extracted.') = True,
+        header: Arg('-H', help='Parse basic data from the Mach-O header.') = False,
+        linked_images: Arg('-K', help='Parse all library images linked by the Mach-O.') = False,
+        signatures: Arg('-S', help='Parse signature and entitlement information.') = False,
+        version: Arg('-V', help="Parse version information from the Mach-O load commands.") = False,
+        load_commands: Arg('-D', help='Parse load commands from the Mach-O header.') = False,
+        exports: Arg('-E', help='List all exported functions.') = False,
+        imports: Arg('-I', help='List all imported functions.') = False,
+        tabular: Arg('-t', help='Print information in a table rather than as JSON') = False,
+    ):
+        super().__init__(
+            header=all or header,
+            linked_images=all or linked_images,
+            version=all or version,
+            signatures=all or signatures,
+            load_commands=load_commands,
+            imports=imports,
+            exports=exports,
+            tabular=tabular,
+        )
+
+    @classmethod
+    def parse_pkcs7_signature(cls, data: bytearray) -> dict:
```

Comment:
I will most likely do that, since I hate code duplication to the point of an unhealthy obsession, but this is nothing you need to concern yourself with.